### DISCUSSION TITLE
Synchronous Generator Method

### DISCUSSION THREAD

**Discussion by tobiasschmidt89** [unocss/unocss#1656] [Q&A] [answered]
Hi there,

does unocss have a synchronous alternative to generate css text to the async `UNOCSSGenerator.generate` method in unocss/core? 

Thanks and kind regards

**Comment by zyyv** [comment, marked as answer]
There doesn't seem to be a synchronous alternative. UnoCSS engine parses into tokens, using `promise.all` to `parseTokens` together. If synchronization is used, the parsing speed may be much slower.

Is there any scruples about not using `await`?

**Reply by tobiasschmidt89** [reply]
Actually I asked due to deno/fresh framework not yet allowing async js for Plug Ins. 

Then I guess I have to wait for deno/fresh to allow async js in Plug Ins.

**Reply by shellscape** [reply]
In a similar situation myself.
